src/index.tsx

```diff
@@ -35,19 +35,16 @@ class App extends React.Component<{}, ICounterState> {
 
     constructor(props: any) {
         super(props);
-        this.button_clicked = this.button_clicked.bind(this);
         this.state = { count: 0 };
     }
 
-    public button_clicked() {
-        this.setState({ count: this.state.count + 1});
-    }
+    private _buttonClicked = () => this.setState({ count: this.state.count + 1});
```

Comment:
`this` hat in JavaScript (mindestens?) 14 Bedeutungen, je nach Kontext. Verwendet man für den Handler innerhalb der Klasse einen Lambda-Ausdruck statt einer Funktion, dann referenziert `this` weiterhin die Klasse und man benötigt das `bind` nicht im Konstruktor. D.h. aber nicht, dass man `bind` nie verwenden sollte. Lambdas eignen sich für kurze Code-Schnipsel. Man muss es im Einzelfall selbst entscheiden.

Ein Handler sollten im Idealfall immer `private` sein. In Javascript gibt es wie gesagt nur `public`, daher wird per Konvention `_` vorangestellt (du wirst diese Schreibweise in vielen öffentlichen Repos finden). Vielleicht würde ich in TypeScript sogar das `_` Prefix einfach weglassen, da man Konventionen leicht vergisst und es deshalb fehlerträchtig ist. Je weniger Konventionen desto besser.

Man könnte die Handler-Funktion außerhalb der Klasse platzieren, um sie aus der public API der Klasse zu entfernen. Dann geht jedoch der `this` Bezug verloren. Ich weiß nicht, ob es möglich ist, dem Handler den `this` Bezug dann als Argument hinzuzufügen. Aber das ist mit TypeScript alles nicht mehr nötig, da es den `private` Modifier gibt.